Makefile

```diff
@@ -26,34 +29,51 @@
 # run all checks (including tests, dialyze and gradualize)
 # > make check
 #
-PROJECT = gradualizer
 
-PLT_APPS = kernel stdlib compiler crypto
-DIALYZER_OPTS = -Werror_handling
+.PHONY: app
+app:
+	rebar3 compile
 
-ESCRIPT_NAME = gradualizer_cli
-ESCRIPT_FILE = $(PROJECT)
+.PHONY: shell
+shell:
+	rebar3 shell
 
-include erlang.mk
+.PHONY: escript
+escript:
+	rebar3 escriptize
+	cp _build/default/bin/gradualizer .
 
+.PHONY: gradualize
 gradualize: escript
 	./gradualizer -pa $(CURDIR)/ebin/ $(CURDIR)/ebin/*beam
-check:: gradualize
 
-# We want warnings to be warnings, not errors.
-ERLC_OPTS := $(filter-out -Werror,$(ERLC_OPTS))
+.PHONY: clean
+clean:
+	rebar3 clean
+	rm -rf _build
+	rm .gradualizer.plt
+	rm gradualizer
+
+.PHONY: tests
+tests:
+	rebar3 eunit
+
+.PHONY: cover
+cover:
+	rebar3 do eunit -c, cover
 
-EUNIT_OPTS = verbose
+.PHONY: showcover
+showcover: cover
+	open _build/test/cover/index.html
 
-# Unit test files to run
-EUNIT_TEST_SOURCES = $(shell ls $(TEST_DIR)/*.erl)
+.PHONY: dialyze
+dialyze:
+	rebar3 dialyzer
 
-# Override erlang.mk variable to only run unit test modules
-# (results in a bit less noisy output)
-# (Other it tried to compile and run all source files
-#  recursively in all subdirs of test)
-EUNIT_TEST_MODS = $(notdir $(basename $(EUNIT_TEST_SOURCES)))
+.PHONY: check
+check: dialyze tests gradualize
```

Comment:
I would prefer first running the tests and then dialyzer, as unit tests are faster.